### ISSUE TITLE
Exposed services should be synced with labels and annotations

### ISSUE THREAD

**Issue by muzcategui1106** [skupperproject/skupper#971]
Skupper exposed services do not have the source service annotations and labels synced. Annotations and labels usually convey important information about objects that other components running in the cluster may depend on. 

Proposal: Have the source service annotations and labels synced to the exposed services in target clusters

I understand this may not be a straight forward task as it may seem. One of the problems that I see is race conditions where multiple skupper instances exposed the same services but have defined different annotations keys or values.

To handle disparity in annotation/label keys, the target service can contain a combination of all the source annotations and labels keys. When it comes to the values, this may need some more in depth thinking.

**Comment by fgiorgetti** [comment]
I am not sure if that helps you, but if you are exposing a service, a deployment or daemonset, you could add the
`skupper.io/service-labels` annotation to it, with a comma separated list of 'label=value', like: 

```
    skupper.io/service-labels: label1=value1,label2=value2
```

But Skupper does not yet have the same for annotations.

If you are using the CLI you could use `skupper service label` to manage the shared labels.

**Comment by muzcategui1106** [comment]
yes confirmed that the service label work. Sadly due to label size limitations, what would really come useful here is the annotation sync

**Comment by fgiorgetti** [comment]
I believe it can be done similarly to the way that Skupper deals with labels (via annotations and CLI).
We just need to ensure that the skupper annotations are not replicated and we should be good.
@grs @ajssmith @nluaces @muzcategui1106 What do you all think?

**Comment by grs** [comment]
Sounds good to me.

**Comment by muzcategui1106** [comment]
Another thing that I found out, is that the label sync just works when I create the service, if I update the annotation to either add a new label or update the value of one of the labels, it does not seem to sync

**Comment by muzcategui1106** [comment]
Just following up on the above comment, were you able to replicate the issue were label sync does not work when the service is updated? it seems to only work when the service is created

**Comment by grs** [comment]
> Just following up on the above comment, were you able to replicate the issue were label sync does not work when the service is updated? 

Yes, and that should now be fixed by https://github.com/skupperproject/skupper/commit/24b1c888799f2fc52c1e6fd616441152e25940cf

**Comment by grs** [comment]
There is now also an alternative way to do this (and it would also bypass the issues in #972), enabled by https://github.com/skupperproject/skupper/commit/46d829dc3c962c7f917d2c0c5dc8fb0ea006fe6d. This allows you do define a service that is annotated only for accepting requests, not as a target for those requests. You can then disable service-sync from all sites, and explicitly control all service creation.

**Comment by muzcategui1106** [comment]
Can you please elaborate on a scenario where it would be beneficial to disable service sync?

**Comment by grs** [comment]
It could be useful where you want to directly control which sites use a skupper service, and more directly specify the labels/annotations on the service used for that. It would avoid the service being removed when connection witg the site in which the target for the service is found was lost.

**Comment by grs** [comment]
@muzcategui1106 regarding the annotations sync, there is a PR for this #985 but at present it follows the approach taken by labels and so would restrict the annotation values you can specify (e.g. cannot include quotes or commas). Does this sound useful as is? Or would you need the ability to set arbitrary annotation values (in which case we would need a different way of specifying them, e.g. the key having some prefix)?
